LibOS/shim/include/shim_fs.h

```diff
@@ -159,8 +175,16 @@ struct shim_d_ops {
     int (*open) (struct shim_handle * hdl, struct shim_dentry * dent,
                  int flags);
 
-    /* look up dentry and allocate internal data */
+    /* look up dentry and allocate internal data.
+     * 
+     * On a successful lookup  (non-error, can be negative),
+     * this function should call get_new_dentry(), populating additional fields,
+     * and storing the new dentry in dent.
+     * 
+     * Maybe drop force?
+     */
```

Comment:
That is helpful information, and would not hurt to document.

I think the new design still does the check once, but does cache negative dentries to avoid this in the future.  There may be a cleaner way to do this.  I found that force led to a lot of weird cases in the state of the cache, that may have been ok, but were harder to reason about.